quadriga/src/main/java/edu/asu/diging/quadriga/web/service/impl/GraphCreationServiceImpl.java

```diff
@@ -0,0 +1,311 @@
+
+        if (sourceURI != null && sourceURI.contains("www.digitalhps.org")) {
+
+            node.setUri(sourceURI);
+            CachedConcept conceptCache = conceptPowerService.getConceptByUri(sourceURI);
+
+            if (conceptCache != null) {
+                node.setLabel(conceptCache.getWord());
+                node.setDescription(conceptCache.getDescription());
+                node.setAlternativeUris(conceptCache.getAlternativeUris());
+            }
+        } else {
+            // Need to get data from viaf if viaf URL is present, but doesn't seem to be part of story Q20-19
+            node.setLabel(event.getTerm().getPrintedRepresentation().getTermParts().iterator().next().getExpression());
+        }
+
+        return node;
+    }
+    
+    /**
+     * Creates a graph node based on the provided TripleElement.
+     *
+     * @param tripleElement  the TripleElement from which to create the node
+     * @param graphNodeType  the type of the graph node (subject, predicate, or object)
+     * @param conceptNodeMap a map containing already created graph nodes, used to avoid duplication
+     * @param nodes          the list of graph nodes to which the newly created node will be added
+     * @return the created GraphNodeData representing the graph node
+     */
+    private GraphNodeData createNode(TripleElement tripleElement, GraphNodeType graphNodeType,
+            Map<String, GraphNodeData> conceptNodeMap, List<GraphData> nodes) {
+        
+        String elementUri = tripleElement.getUri();
+        //Avoid node duplication if the element is not a predicate element and if it already exists
+        if (graphNodeType != GraphNodeType.PREDICATE && elementUri != null && !elementUri.isEmpty()
+                && conceptNodeMap.containsKey(elementUri)) {
+            return conceptNodeMap.get(elementUri);
+        }
+        GraphNodeData nodeData = new GraphNodeData();
+        nodeData.setId(new ObjectId().toString());
+        nodeData.setLabel(tripleElement.getLabel());
+        nodeData.setGroup(graphNodeType.getGroupId());
+        nodeData.setUri(tripleElement.getUri());
+        if (graphNodeType != GraphNodeType.PREDICATE && elementUri != null && !elementUri.isEmpty()) {
+            conceptNodeMap.put(elementUri, nodeData);
+        }
+        nodes.add(nodeData);
+        return nodeData;
+    }
+    
+    /**
+     * Creates that edge that links the provided source and target using their IDs and it to
+     * the list of graphEdges
+     * 
+     * @param graphEdges is the list that maintains all edges created for the graph
+     * @param sourceId is the source node's id to be linked to the target
+     * @param targetId is the target node's id to be linked to the source
+     * @param eventGraphId is the id of the event graph.
+     */
+    private void createEdge(List<GraphData> graphEdges, String sourceId, String targetId, String eventGraphId) {
```

Comment:
is this used anywhere else? It looks like it's only called with `eventGraphId` being null? If so, why is the parameter there?

Reply:
Yes its called in the same file on line 118, 122 where we create an edge between subject node and predicate node, and between object node and predicate node respectively.Here we add the evenGraphId to theGraphEdgeData object created.